### PR TITLE
watchOS

### LINKED ISSUE #15: watchOS support

### PR DESCRIPTION
**Pull request by bassrock**
This seems to work and fix #15, The same approach can also be used to add watchUI classes and OS X UI Classes.

### CODE REVIEW

**Comment by kean** [comment]
Hi @bassrock, thanks for the pull request! I'll take a look into it as soon as I can.

From what I saw at the first glance:
1. Please branch of the develop branch instead of master
2. Instead of creating a different subspec with UI components for each platform I would rather keep 'UI' subspec and specify [source files per platform](https://guides.cocoapods.org/syntax/podspec.html#group_multi_platform_support):

``` ruby
    s.subspec "UI" do |ss|
        ss.dependency "DFImageManager/Core"
        ss.source_files = "DFImageManager/Source/UI/**/*.{h,m}" # shared files
        ss.ios.source_files = "DFImageManager/Source/UI/iOS/*.{h,m}" # platform specific files
    end
```

File: DFImageManager/Source/Core/Caching/DFImageCache.m [hunk 1]
```diff
@@ -52,7 +55,7 @@ - (instancetype)init {
 - (nullable DFCachedImageResponse *)cachedImageResponseForKey:(nullable id<NSCopying>)key {
     DFCachedImageResponse *response = [_cache objectForKey:key];
     if (response) {
-        if (response.expirationDate > CACurrentMediaTime()) {
+        if (response.expirationDate > mach_absolute_time()) {
```
Comment (kean) [inline DFImageManager/Source/Core/Caching/DFImageCache.m]: You can't use mach_absolute_time() like that. It returns time in processing units, not seconds like CACurrentMediaTime(). Conversion between mach_absolute_time() units and seconds is quite complicated.

Anyway, both functions are useful when you need to measure time in terms of nanoseconds, which is not the case here. NSDate looks like a good candidate, but it's rather heavy and this code needs to run on the main thread. I would rather use something that is faster.

Comment (bassrock) [inline DFImageManager/Source/Core/Caching/DFImageCache.m]: Yea the reasoning here is that CACurrentMediaTime is not available on the watch. Any suggestions for a replacement works.

**Comment by bassrock** [comment]
Ok will update the specs in a bit. Also this is branched off of develop.

**Comment by bassrock** [comment]
I think CFAbsoluteTimeGetCurrent works.

**Comment by kean** [comment]
CFAbsoluteTimeGetCurrent is ok :+1:

Subspec name is still iOSUI `s.subspec "iOSUI" do |ss|`. Could you please rename it back to "UI" and I think it would be ok to merge this?

**Comment by kean** [comment]
Merging, thanks for your contribution @bassrock  :+1:
P.S. If you want to submit more, please branch of the `develop` branch and create pull requests for the develop branch, thanks :)

**Comment by kean** [comment]
I've removed entire xcodeproj with framework targets and moved to classic CocoaPod project structure with a single Xcode project. Now there is no need to manage two build configuration at the same time, we rely on a single build configuration, which is podpsec and CocoaPods.

**Pull request merged**